library/Ivoz/Provider/Infrastructure/Domain/Service/Company/SendUsersTrustedPermissionsReloadRequest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Ivoz\Provider\Infrastructure\Domain\Service\Company;
+
+use Ivoz\Kam\Domain\Service\UsersClientInterface;
+use Ivoz\Provider\Domain\Model\Company\CompanyInterface;
+use Ivoz\Provider\Domain\Service\Company\CompanyLifecycleEventHandlerInterface;
+
+class SendUsersTrustedPermissionsReloadRequest implements CompanyLifecycleEventHandlerInterface
+{
+    const ON_COMMIT_PRIORITY = self::PRIORITY_NORMAL;
+
+    protected $usersClient;
+
+    public function __construct(
+        UsersClientInterface $usersClient
+    ) {
+        $this->usersClient = $usersClient;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            self::EVENT_ON_COMMIT => self::ON_COMMIT_PRIORITY
+        ];
+    }
+
+    public function execute(CompanyInterface $company)
+    {
+        $wasRemoved = ($company->hasChanged('id') && !$company->getId());
```

Comment:
This could be also implemented in a future like `isNew()` method? `isDeleted()` maybe?

Reply:
Actually we've hasBeenDeleted method :)